Core/src/Detector/CuboidalContainerBuilder.cpp

```diff
@@ -0,0 +1,177 @@
+    throw std::invalid_argument(
+        "CuboidalContainerBuilder: Invalid binning value. Only Acts::binX, "
+        "Acts::binY, Acts::binZ are supported.");
+  }
+
+  m_cfg.auxiliary = "*** acts auto-generated from proxy ***";
+  m_cfg.geoIdGenerator = bpNode.geoIdGenerator;
+  m_cfg.rootVolumeFinderBuilder = bpNode.rootVolumeFinderBuilder;
+}
+
+Acts::Experimental::DetectorComponent
+Acts::Experimental::CuboidalContainerBuilder::construct(
+    const GeometryContext& gctx) const {
+  // Return container object
+  DetectorComponent::PortalContainer rContainer;
+  bool atNavigationLevel = true;
+
+  // Create the indivudal components, collect for both outcomes
+  std::vector<DetectorComponent> components;
+  ACTS_DEBUG("Building container from " << m_cfg.builders.size()
+                                        << " components.");
+  // Check through the component volumes - if every builder only
+  // built exactly one volume, you are at pure navigation level
+  // Collect the volumes
+  std::vector<std::shared_ptr<DetectorVolume>> volumes;
+  std::vector<DetectorComponent::PortalContainer> containers;
+  std::vector<std::shared_ptr<DetectorVolume>> rootVolumes;
+  // Run through the builders
+  std::for_each(
+      m_cfg.builders.begin(), m_cfg.builders.end(), [&](const auto& builder) {
+        auto [cVolumes, cContainer, cRoots] = builder->construct(gctx);
+        atNavigationLevel = (atNavigationLevel && cVolumes.size() == 1u);
+        ACTS_VERBOSE("Number of volumes: " << cVolumes.size());
+        // Collect individual components, volumes, containers, roots
+        volumes.insert(volumes.end(), cVolumes.begin(), cVolumes.end());
+        containers.push_back(cContainer);
+        rootVolumes.insert(rootVolumes.end(), cRoots.volumes.begin(),
+                           cRoots.volumes.end());
+      });
+  // Navigation level detected, connect volumes (cleaner and faster than
+  // connect containers)
+  if (atNavigationLevel) {
+    ACTS_VERBOSE(
+        "Component volumes are at navigation level: connecting volumes.");
+    // Connect volumes
+    rContainer = Acts::Experimental::detail::CuboidalDetectorHelper::connect(
+        gctx, volumes, m_cfg.binning, {}, logger().level());
+
+  } else {
+    ACTS_VERBOSE("Components contain sub containers: connect containers.");
+    // Connect containers
+    rContainer = Acts::Experimental::detail::CuboidalDetectorHelper::connect(
+        gctx, containers, m_cfg.binning, {}, logger().level());
+  }
+  ACTS_VERBOSE("Number of root volumes: " << rootVolumes.size());
+
+  // Check if a root volume finder is provided
+  if (m_cfg.rootVolumeFinderBuilder) {
+    // Return the container
+    return Acts::Experimental::DetectorComponent{
+        volumes, rContainer,
+        RootDetectorVolumes{
+            rootVolumes,
+            m_cfg.rootVolumeFinderBuilder->construct(gctx, rootVolumes)}};
+  }
+
+  // Geometry Id generation
+  if (m_cfg.geoIdGenerator != nullptr) {
```

Comment:
I think this block somehow should be before the root Volume finder, because in case a root volume finder is provided, it looks like we return without GeometryID setting ?

I this the case in the `CylindricalContainerBuilder` as well ?

Is so it's a bug, and we should fix it in both.

Reply:
Yes, looks like geometry id is not going to be assigned. 

The same thing is happening in the CylindricalContainerBuilder. 

Will push the fix.